community/learn/graphql-tutorials/tutorials/react-apollo-hooks/app-final/src/components/Todo/TodoPublicList.js

```diff
@@ -0,0 +1,132 @@
+import React, { Fragment, useState, useEffect } from 'react';
+import { useSubscription, useApolloClient } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+import TaskItem from './TaskItem';
+
+const TodoPublicList = (props) => {
+  const [state, setState] = useState({
+    olderTodosAvailable: props.latestTodo ? true : false,
+    newTodosCount: 0,
+    error: false,
+    todos: [],
+    oldestTodoId: props.latestTodo ? (props.latestTodo.id + 1) : 0,
+    newestTodoId: props.latestTodo ? props.latestTodo.id : 0, 
+  });
+  
+  const client = useApolloClient();
+
+  useEffect(() => {
+    loadOlder();
+  }, []);
+
+  useEffect(() => {
+    if (props.latestTodo && props.latestTodo.id > state.newestTodoId) {
+      setState({...state, newestTodoId: props.latestTodo.id, newTodosCount: state.newTodosCount + 1});
```

Comment:
This a hook setState so not sure how exactly can callback style help here since that callback would anyway be needed to be in useEffect hook. Let me know what should be done.

Reply:
https://reactjs.org/docs/hooks-reference.html#functional-updates

Reply:
Will look into it. Thanks!

Reply:
Done!